client/components/d3chart/utils.js

```diff
@@ -458,22 +456,12 @@ export const drawAxis = ( node, params ) => {
 		.remove();
 };
 
-const getTooltipRowLabel = ( d, row, params ) =>
-	d[ row.key ].labelDate ? formatDate( params.pointLabelFormat, d[ row.key ].labelDate ) : row.key;
-
-const getFormatedTotal = ( total, valueType ) => {
-	let rowTotal = total;
-	switch ( valueType ) {
-		case 'average':
-			rowTotal = Math.round( total );
-			break;
-		case 'currency':
-			rowTotal = formatCurrency( total );
-			break;
-		case 'number':
-			break;
-	}
-	return rowTotal;
+const getTooltipRowLabel = ( d, row, params ) => {
+	const isDate =
+		d[ row.key ].labelDate instanceof Date
+			? d[ row.key ].labelDate
+			: new Date( d[ row.key ].labelDate );
+	return d[ row.key ].labelDate ? params.tooltipLabelFormat( isDate ) : row.key;
```

Comment:
I'm not sure to understand how this function works. Couldn't we check if `labelDate` exists before doing everything else?
```ES6
const getTooltipRowLabel = ( d, row, params ) => {
	if ( d[ row.key ].labelDate ) {
		const isDate = d[ row.key ].labelDate instanceof Date
			? d[ row.key ].labelDate
			: new Date( d[ row.key ].labelDate );
		return params.tooltipLabelFormat( isDate );
	}
	return row.key;
};
```